File: taskManager/labfinal_210042111/CommandLogger.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace labfinal_210042111
{
    public class CommandLogger:IObserver
    {
        private List<string> logHistory = new List<string>();
      
        public void Update(string message)
        {
            logHistory.Add(message);
        }
        public List<string> getLogHistory()
        {
            return logHistory;
        }
    }
}

File: TaskManager/src/CommandLogger.java
import java.util.ArrayList;
import java.util.List;
public class CommandLogger implements Observer {
    private List<String> logHistory = new ArrayList<>();
    @Override
    public void update(String message) {
        logHistory.add(message);
    }
    public List<String> getLogHistory() {
        return logHistory;
    }
}